### ISSUE TITLE
TCP stats improvements

### ISSUE THREAD

**Issue by cmunger** [epickrram/angler#11]
Hello, played with TCP stats on the 1.5 release and found out a few things :

First issue with tcp6 support lack :
Currently you support only tcp4 /proc/net/tcp, however java by default uses tcp6 stack which registers connections in /proc/net/tcp6 so we cannot monitor anything with your lib unless we specify the -Djava.net.preferIPv4Stack=true to force java using tcp4 stack, this is a big no I guess if you want to use that in prod.

Second issue with TcpSocketMonitor api : 
There are methods to monitor a socket using only one InetSocketAdress as parameter :
public void beginMonitoringOf(final InetSocketAddress socketAddress) {}

This causes problems when you try to use this lib with a serversocket where you need the pair of local and remote address to monitor correctly a single tcp connection (they provide both local and remote socket info in /proc/net/tcp), so you should add to your API a method like public void beginMonitoringOf(final InetSocketAddress localAddress,final InetSocketAddress remoteAddress ) {}
You should even i think remove the API method with a single InetSocketAddress parameter.

I guess that unless you implement these 2 functionalities the TCP implementation is not usable which is sad since your lib is quite cool :) thanks for the good job.

**Comment by epickrram** [comment]
Hello,
thanks for your comments. Firstly, ipv6 is not yet supported (I will make this more obvious on the readme), but I plan to add this functionality soon.

At LMAX, we run all our systems with preferIPv4Stack=true, so we haven't really had the need for it yet. There are a couple of changes needed to the core of Angler to enable this to happen, which will be addressed soon.

On TcpSocketMonitor usage, this should work OK for local sockets. For instance, I can do the following:

```
[mark@localhost ~]$ nc -l 127.0.0.1 4096
[mark@localhost ~]$ cat /proc/net/tcp | head
  sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt...
   2: 0100007F:1000 00000000:0000 0A 00000000:00000000 00:00000000...

```

and use Angler to monitor the buffer depths of the 127.0.0.1:4096 socket. Admittedly, we only ever consider the local address (this fits our use-case better), but I can see that providing the complete local/remote pair may make things clearer.

I won't remove the single InetAddress API calls, since we often want to monitor a socket that is not owned or created by our code, and in this case, we only know the local address.

Adding the local/remote pair subscription call will require the internal changes needed by IPv6 support, so I will address this once IPv6 is in place.

Closing this in favour of #12 and #5

**Comment by cmunger** [comment]
Hello, thanks for the feedback, I'll give another try to your lib once ipv6 is implemented and local/remote addr API in place. The other trick with ipv6 support will be to see if you have to look in /proc/net/tcp or /proc/net/tcp6, i guess that if the preferIPv4Stack=true is not set then go by default in /proc/net/tcp6. Regards
